inc/namespace.php

```diff
@@ -410,3 +413,46 @@ function real_url_path( ?string $url, string $handle ) : ?string {
 function clear_signups_cache() {
 	wp_cache_set( 'last_changed', microtime(), 'signups' );
 }
+
+/**
+ * Determines whether a redirect should actucally occur.
+ *
+ * In some instances redirect_canonical() will rebuild the query string which can change
+ * its formatting by removing trailing = signs and URL encoding keys. This can result in
+ * a redirect that will generate the same Batcache key. Once the redirect is cached then
+ * Batcache will start to redirect endlessly. This filter prevents that behaviour.
+ *
+ * @param string $redirect_url The URL being redirected to.
+ * @param string $requested_url The original URL requested.
+ * @return string The filtered redirect target URL.
+ */
+function maybe_redirect( $redirect_url, $requested_url ) {
+	$redirect_query_string = parse_url( $redirect_url, PHP_URL_QUERY );
+	$requested_query_string = parse_url( $redirect_url, PHP_URL_QUERY );
```

Comment:
@roborourke I'm investigating a similar bug to this, I'm not sure it's exactly the same. However, I noticed this code looks wrong? You are calling `parse_url()` on the same variable each time, when I assume it's meant to be on `redirect_url` and `requested_url`?

Reply:
Yeah that does look wrong, thanks for flagging!